internal/xds/rbac/rbac_engine_test.go

```diff
@@ -0,0 +1,625 @@
+					evaluateArgs: &RPCData{
+						PeerInfo: &peer.Peer{
+							Addr: &addr{ipAddress: "0.0.0.0"},
+						},
+					},
+					wantMatchingPolicyName: "certain-source-ip",
+					matchingPolicyFound:    true,
+				},
+				// This incoming RPC Call shouldn't match with the certain-source-ip policy.
+				{
+					evaluateArgs: &RPCData{
+						PeerInfo: &peer.Peer{
+							Addr: &addr{ipAddress: "10.0.0.0"},
+						},
+					},
+					wantMatchingPolicyName: "",
+					matchingPolicyFound:    false,
+				},
+			},
+		},
+		{
+			name: "TestDestinationIpMatcher",
+			rbacConfig: &v3rbacpb.RBAC{
+				Policies: map[string]*v3rbacpb.Policy{
+					"certain-destination-ip": {
+						Permissions: []*v3rbacpb.Permission{
+							{Rule: &v3rbacpb.Permission_DestinationIp{DestinationIp: &v3corepb.CidrRange{AddressPrefix: "0.0.0.0", PrefixLen: &wrapperspb.UInt32Value{Value: uint32(10)}}}},
+						},
+						Principals: []*v3rbacpb.Principal{
+							{Identifier: &v3rbacpb.Principal_Any{Any: true}},
+						},
+					},
+				},
+			},
+			rbacQueries: []struct {
+				evaluateArgs           *RPCData
+				wantMatchingPolicyName string
+				matchingPolicyFound    bool
+			}{
+				// This incoming RPC Call should match with the certain-destination-ip policy.
+				{
+					evaluateArgs: &RPCData{
+						DestinationAddr: &addr{ipAddress: "0.0.0.10"},
+					},
+					wantMatchingPolicyName: "certain-destination-ip",
+					matchingPolicyFound:    true,
+				},
+				// This incoming RPC Call shouldn't match with the certain-destination-ip policy.
+				{
+					evaluateArgs: &RPCData{
+						DestinationAddr: &addr{ipAddress: "10.0.0.0"},
+					},
+					wantMatchingPolicyName: "",
+					matchingPolicyFound:    false,
+				},
+			},
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			// Instantiate the rbacEngine with different configurations that
+			// interesting to test and to query.
+			rbacEngine, err := NewEngine(test.rbacConfig)
+			if err != nil {
+				t.Fatalf("Error constructing RBAC Engine: %v", err)
+			}
+			// Query that created RBAC Engine with different args to see if the
+			// RBAC Engine configured a certain way works as intended.
+			for _, queryToRBACEngine := range test.rbacQueries {
+				// The matchingPolicyName returned will be empty in the case of
+				// no matching policy. Thus, matchingPolicyName can also be used
+				// to test the "error" condition of no matching policies.
+				matchingPolicyName, matchingPolicyFound := rbacEngine.FindMatchingPolicy(queryToRBACEngine.evaluateArgs)
+				if matchingPolicyFound != queryToRBACEngine.matchingPolicyFound {
+					t.Fatalf("Got matchingPolicyFound bool: %v, want matchingPolicyFound: %v", matchingPolicyFound, queryToRBACEngine.matchingPolicyFound)
+				}
+				if matchingPolicyName != queryToRBACEngine.wantMatchingPolicyName {
+					t.Fatalf("Got matching policy name: %v, want matching policy name: %v", matchingPolicyName, queryToRBACEngine.wantMatchingPolicyName)
+				}
+			}
```

Comment:
Combine the two checks into the same `if`. Or, change the calls to `t.Fatalf` to `t.Errorf` because we want to know if both of them match expected values or not.

Also, please identify the function under test and its args in the error message.
`t.Errorf("FindMatchingPolicy(%+v) returned (%v, %v), want (%v, %v)", ....)`

Reply:
Done.